docroot/modules/custom/va_gov_benefits/src/EventSubscriber/BenefitsSubscriber.php

```diff
@@ -0,0 +1,115 @@
+
+use Drupal\core_event_dispatcher\Event\Form\FormAlterEvent;
+use Drupal\core_event_dispatcher\FormHookEvents;
+use Drupal\Core\Entity\ContentEntityForm;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\va_gov_user\Service\UserPermsService;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * VA.gov VA Benefits Entity Event Subscriber.
+ */
+class BenefitsSubscriber implements EventSubscriberInterface {
+  use StringTranslationTrait;
+
+  /**
+   * The User Perms Service.
+   *
+   * @var \Drupal\va_gov_user\Service\UserPermsService
+   */
+  protected $userPermsService;
+
+  /**
+   * Constructs the EventSubscriber object.
+   *
+   * @param \Drupal\va_gov_user\Service\UserPermsService $user_perms_service
+   *   The user perms service.
+   */
+  public function __construct(
+    UserPermsService $user_perms_service
+  ) {
+    $this->userPermsService = $user_perms_service;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents(): array {
+    return [FormHookEvents::FORM_ALTER => 'formAlter'];
+  }
+
+  /**
+   * Form alter Event call.
+   *
+   * @param \Drupal\core_event_dispatcher\Event\Form\FormAlterEvent $event
+   *   The event.
+   */
+  public function formAlter(FormAlterEvent $event): void {
+    $form = &$event->getForm();
+    $form_state = $event->getFormState();
+    if (!empty($form["#form_id"]) && $form["#form_id"] === 'taxonomy_term_va_benefits_taxonomy_form') {
+      $this->lockApiIdEditing($form, $form_state);
+      $this->lockBenefitName($form, $form_state);
+      $this->renameTitleLabelAndDescription($form);
+      $this->hideRelations($form);
+    }
+  }
+
+  /**
+   * Locks down API Id for non-admins.
+   *
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public function lockApiIdEditing(array &$form, FormStateInterface $form_state) {
+    $form_object = $form_state->getFormObject();
+    $bundle = NULL;
+    if ($form_object instanceof ContentEntityForm) {
+      $bundle = $form_object->getEntity()->bundle();
+    }
+    if (!$this->userPermsService->hasAdminRole(TRUE) && $bundle === 'va_benefits_taxonomy') {
+      $form['field_va_benefit_api_id']['#disabled'] = TRUE;
+    }
+  }
+
+  /**
+   * Locks down Official Benefit name for non-admins.
+   *
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public function lockBenefitName(array &$form, FormStateInterface $form_state) {
+    $form_object = $form_state->getFormObject();
+    $bundle = NULL;
+    if ($form_object instanceof ContentEntityForm) {
+      $bundle = $form_object->getEntity()->bundle();
+    }
+    if (!$this->userPermsService->hasAdminRole(TRUE) && $bundle === 'va_benefits_taxonomy') {
+      $form['name']['#disabled'] = TRUE;
+    }
```

Comment:
Could:  reduce lines of code and unnecesary checks.
```suggestion
    if ($form_object instanceof ContentEntityForm) {
      $bundle = $form_object->getEntity()->bundle();
      if (!$this->userPermsService->hasAdminRole(TRUE) && $bundle === 'va_benefits_taxonomy') {
        $form['name']['#disabled'] = TRUE;
      }
    }
```